test/cql-pytest/test_json.py

```diff
@@ -136,6 +136,31 @@ def test_fromjson_bigint_overflow_prepared(cql, table1):
     stmt = cql.prepare(f"INSERT INTO {table1} (p, bigv) VALUES (?, fromJson(?))")
     with pytest.raises(FunctionFailure):
         cql.execute(stmt, [p, '9223372036854775808'])
+def test_fromjson_bigint_float_overflow_prepared(cql, table1):
+    # -9223372036854775808
+    MIN_BIGINT = -(2 ** 63)
+    # +9223372036854775807
+    MAX_BIGINT = 2 ** 63 - 1
+    p = unique_key_int()
+    stmt = cql.prepare(f"INSERT INTO {table1} (p, bigv) VALUES (?, fromJson(?))")
+    # all the tested number in this test case can be represented by IEEE-754
+    # 64-bit float, in order to verify that we only accept the numbers in the
+    # range of bigint without worrying if RapidJSON nearest rounds them first.
+    with pytest.raises(FunctionFailure):
+        # the last IEEE-754 64-bit float point number less than min value
+        # of 64-bit signed long
+        cql.execute(stmt, [p, f'{MIN_BIGINT - 2048}.0'])
```

Comment:
Did this test **fail** before your patch? If so, maybe I'm finally starting to understand what you tried to fix in this patch! I kept thinking it was about numbers very close to the cutoff, but this test is about numbers more far away from the cutoff and should more obviously result in an error.

If the answer is yes, please add a comment before this test explaining that it reproduces issue #NNNN, otherwise I can only guess how this specific tests relates to the issue or your fix.

Reply:
Seeing that you have here a bunch of checks one after another, it will also be helpful to have a comment "reproduces issue #NNNN" above the specific checks that failed before the patch. I assume it's not all of them?